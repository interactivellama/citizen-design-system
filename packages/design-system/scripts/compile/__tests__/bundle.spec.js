// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */

const Task = require("data.task");
const Bundle = require("../bundle");

it("doesn't change configs", () => {
  expect.assertions(1);
  Bundle.configs
    .map(cfgs =>
      cfgs.map(c => c.set("plugins", null).deleteIn(["output", "path"]))
    )
    .fork(
      e => {
        throw e;
      },
      cfgs => {
        expect(cfgs.toJS()).toMatchSnapshot();
      }
    );
});
