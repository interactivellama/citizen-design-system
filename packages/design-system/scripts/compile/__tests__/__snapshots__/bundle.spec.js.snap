// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't change configs 1`] = `
"[
  {
    \\"mode\\": \\"production\\",
    \\"module\\": {
      \\"rules\\": [
        {
          \\"exclude\\": {
          },
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"thread-loader\\",
              \\"options\\": {
              }
            },
            {
              \\"loader\\": \\"babel-loader\\",
              \\"options\\": {
                \\"babelrc\\": false,
                \\"plugins\\": [
                  \\"transform-object-rest-spread\\"
                ],
                \\"presets\\": [
                  \\"es2015\\",
                  \\"react\\"
                ]
              }
            }
          ]
        },
        {
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"style-loader\\"
            },
            {
              \\"loader\\": \\"raw-loader\\"
            },
            {
              \\"loader\\": \\"sass-loader\\"
            }
          ]
        },
        {
          \\"exclude\\": {
          },
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"thread-loader\\",
              \\"options\\": {
              }
            },
            {
              \\"loader\\": \\"babel-loader\\",
              \\"options\\": {
                \\"babelrc\\": false,
                \\"plugins\\": [
                  \\"transform-object-rest-spread\\"
                ],
                \\"presets\\": [
                  \\"es2015\\",
                  \\"react\\"
                ]
              }
            },
            {
              \\"loader\\": \\"/scripts/compile/mdx-loader.js\\"
            }
          ]
        }
      ]
    },
    \\"optimization\\": {
    },
    \\"output\\": {
      \\"filename\\": \\"__internal/slds.umd.js\\",
      \\"globalObject\\": \\"this\\",
      \\"library\\": \\"SLDS\\",
      \\"libraryTarget\\": \\"umd\\",
      \\"path\\": \\"/.dist\\",
      \\"publicPath\\": \\"/assets/scripts/bundle/\\"
    },
    \\"resolve\\": {
      \\"alias\\": {
        \\"lodash\\": \\"\\"
      },
      \\"extensions\\": [
        \\".js\\",
        \\".jsx\\"
      ]
    }
  },
  {
    \\"externals\\": {
      \\"js-beautify\\": \\"JSBeautify\\",
      \\"react\\": \\"React\\",
      \\"react-dom\\": \\"ReactDOM\\"
    },
    \\"mode\\": \\"production\\",
    \\"module\\": {
      \\"rules\\": [
        {
          \\"exclude\\": {
          },
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"thread-loader\\",
              \\"options\\": {
              }
            },
            {
              \\"loader\\": \\"babel-loader\\",
              \\"options\\": {
                \\"babelrc\\": false,
                \\"plugins\\": [
                  \\"transform-object-rest-spread\\"
                ],
                \\"presets\\": [
                  \\"es2015\\",
                  \\"react\\"
                ]
              }
            }
          ]
        },
        {
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"style-loader\\"
            },
            {
              \\"loader\\": \\"raw-loader\\"
            },
            {
              \\"loader\\": \\"sass-loader\\"
            }
          ]
        },
        {
          \\"exclude\\": {
          },
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"thread-loader\\",
              \\"options\\": {
              }
            },
            {
              \\"loader\\": \\"babel-loader\\",
              \\"options\\": {
                \\"babelrc\\": false,
                \\"plugins\\": [
                  \\"transform-object-rest-spread\\"
                ],
                \\"presets\\": [
                  \\"es2015\\",
                  \\"react\\"
                ]
              }
            },
            {
              \\"loader\\": \\"/scripts/compile/mdx-loader.js\\"
            }
          ]
        }
      ]
    },
    \\"optimization\\": {
      \\"splitChunks\\": {
        \\"chunks\\": \\"all\\",
        \\"minChunks\\": 2,
        \\"name\\": \\"__internal/chunked/showcase/common.js\\"
      }
    },
    \\"output\\": {
      \\"filename\\": \\"[name]\\",
      \\"globalObject\\": \\"this\\",
      \\"jsonpFunction\\": \\"webpackJsonpSLDS___internal_chunked_showcase\\",
      \\"library\\": [
        \\"SLDS\\",
        \\"[name]\\"
      ],
      \\"path\\": \\"/.dist\\",
      \\"publicPath\\": \\"/assets/scripts/bundle/\\"
    },
    \\"resolve\\": {
      \\"alias\\": {
        \\"lodash\\": \\"\\"
      },
      \\"extensions\\": [
        \\".js\\",
        \\".jsx\\"
      ]
    }
  },
  {
    \\"externals\\": {
      \\"js-beautify\\": \\"JSBeautify\\",
      \\"react\\": \\"React\\",
      \\"react-dom\\": \\"ReactDOM\\"
    },
    \\"mode\\": \\"production\\",
    \\"module\\": {
      \\"rules\\": [
        {
          \\"exclude\\": {
          },
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"thread-loader\\",
              \\"options\\": {
              }
            },
            {
              \\"loader\\": \\"babel-loader\\",
              \\"options\\": {
                \\"babelrc\\": false,
                \\"plugins\\": [
                  \\"transform-object-rest-spread\\"
                ],
                \\"presets\\": [
                  \\"es2015\\",
                  \\"react\\"
                ]
              }
            }
          ]
        },
        {
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"style-loader\\"
            },
            {
              \\"loader\\": \\"raw-loader\\"
            },
            {
              \\"loader\\": \\"sass-loader\\"
            }
          ]
        },
        {
          \\"exclude\\": {
          },
          \\"test\\": {
          },
          \\"use\\": [
            {
              \\"loader\\": \\"thread-loader\\",
              \\"options\\": {
              }
            },
            {
              \\"loader\\": \\"babel-loader\\",
              \\"options\\": {
                \\"babelrc\\": false,
                \\"plugins\\": [
                  \\"transform-object-rest-spread\\"
                ],
                \\"presets\\": [
                  \\"es2015\\",
                  \\"react\\"
                ]
              }
            },
            {
              \\"loader\\": \\"/scripts/compile/mdx-loader.js\\"
            }
          ]
        }
      ]
    },
    \\"optimization\\": {
      \\"splitChunks\\": {
        \\"chunks\\": \\"all\\",
        \\"minChunks\\": 2,
        \\"name\\": \\"__internal/chunked/docs/common.js\\"
      }
    },
    \\"output\\": {
      \\"filename\\": \\"[name]\\",
      \\"globalObject\\": \\"this\\",
      \\"jsonpFunction\\": \\"webpackJsonpSLDS___internal_chunked_docs\\",
      \\"library\\": [
        \\"SLDS\\",
        \\"[name]\\"
      ],
      \\"path\\": \\"/.dist\\",
      \\"publicPath\\": \\"/assets/scripts/bundle/\\"
    },
    \\"resolve\\": {
      \\"alias\\": {
        \\"lodash\\": \\"\\"
      },
      \\"extensions\\": [
        \\".js\\",
        \\".jsx\\"
      ]
    }
  }
]"
`;
